flytekit/core/type_engine.py

```diff
@@ -969,22 +969,46 @@ def get_literal_type(self, t: Type[T]) -> Optional[LiteralType]:
         except Exception as e:
             raise ValueError(f"Type of Generic List type is not supported, {e}")
 
+    @staticmethod
+    def is_batchable(t: Type):
+        from flytekit.types.pickle import FlytePickle
+
+        if t == FlytePickle or (hasattr(t, "__origin__") and t.__origin__ == FlytePickle):
+            return True
+        if get_origin(t) is not None:
+            return any(map(ListTransformer.is_batchable, get_args(t)))
+        return False
+
     def to_literal(self, ctx: FlyteContext, python_val: T, python_type: Type[T], expected: LiteralType) -> Literal:
+        from flytekit.types.pickle import FlytePickle
+
         if type(python_val) != list:
             raise TypeTransformerFailedError("Expected a list")
 
-        t = self.get_sub_type(python_type)
-        lit_list = [TypeEngine.to_literal(ctx, x, t, expected.collection_type) for x in python_val]  # type: ignore
+        if self.is_batchable(python_type):
+            batchSize = len(python_val)  # default batch size
+            # parse annotated to get the number of items saved in a pickle file.
+            if get_origin(python_type) is Annotated:
+                batchSize = get_args(python_type)[1]
```

Comment:
we should check `get_args(python_type)[1]` is int here because we may have an output type that looks like this annotated[list, [HashMethod](https://docs.flyte.org/projects/flytekit/en/latest/generated/flytekit.HashMethod.html#flytekit.HashMethod)(...)]

Reply:
https://docs.flyte.org/projects/cookbook/en/latest/auto/core/flyte_basics/task_cache.html#caching-of-non-flyte-offloaded-objects

Reply:
Thank you!